controller/src/main/java/org/jboss/as/controller/operations/global/ReadConfigAsFeaturesOperationHandler.java

```diff
@@ -0,0 +1,382 @@
+                if (failureRef.get() != null) {
+                    // One of our subsystems failed
+                    context.getFailureDescription().set(failureRef.get());
+                    failed = true;
+                }
+                if (!failed) {
+                    final List<ModelNode> children = results.asList();
+                    if (feature == null) {
+                        for (ModelNode child : children) {
+                            result.add(child);
+                        }
+                    } else if (nest) {
+                        if (!children.isEmpty()) {
+                            final ModelNode nested = feature.get(ModelDescriptionConstants.CHILDREN).setEmptyList();
+                            for (final ModelNode childRsp : children) {
+                                nested.add(childRsp);
+                            }
+                        }
+                        result.add(feature);
+                    } else {
+                        result.add(feature);
+                        if (!children.isEmpty()) {
+                            for (final ModelNode childRsp : children) {
+                                result.add(childRsp);
+                            }
+                        }
+                    }
+                    context.getResult().set(result);
+                }
+            }
+        }, OperationContext.Stage.MODEL, true);
+
+        describeChildren(resource, registration, address, context, failureRef, results, operation, nest);
+    }
+
+    private void describeChildren(final Resource resource, final ImmutableManagementResourceRegistration registration,
+            final PathAddress address, OperationContext context,
+            final AtomicReference<ModelNode> failureRef, final ModelNode results, ModelNode operation, Boolean nest) {
+        resource.getChildTypes().forEach((childType) -> {
+            resource.getChildren(childType).stream()
+                    .filter(entry -> {
+                        final ImmutableManagementResourceRegistration childRegistration = registration.getSubModel(PathAddress.EMPTY_ADDRESS.append(entry.getPathElement()));
+                        if(childRegistration == null) {
+                            ControllerLogger.ROOT_LOGGER.warnf("Couldn't find a registration for %s at %s for resource %s at %s", entry.getPathElement().toString(), registration.getPathAddress().toCLIStyleString(), resource, address.toCLIStyleString());
+                            return false;
+                        }
+                        return !childRegistration.isRuntimeOnly() && !childRegistration.isRemote() && !childRegistration.isAlias();
+                    })
+                    .filter(entry ->  addressFilter == null || addressFilter.accepts(address.append(entry.getPathElement())))
+                    .forEach((entry) -> describeChildResource(entry, registration, address, context, failureRef, results, operation, nest));
```

Comment:
Our general coding standard is to not use streams to replace for loops as the lambdas add a lot to our memory footprint compared to normal loop content.